python/cudf/cudf/core/dataframe.py

```diff
@@ -6533,12 +6533,14 @@ def to_struct(self, name=None):
         field_names = [str(name) for name in self._data.names]
 
         col = cudf.core.column.build_struct_column(
-            names=field_names, children=self._data.columns, size=len(self)
+            names=field_names,
+            children=tuple(
+                [col.copy(deep=True) for col in self._data.columns]
+            ),
+            size=len(self),
         )
         return cudf.Series._from_data(
-            cudf.core.column_accessor.ColumnAccessor(
-                {name: col.copy(deep=True)}
-            ),
```

Comment:
Or did it happen here? If it did, why do we need to make the change to explicitly copy the children? Does `col.copy(deep=True)` not copy children? If so that feels like a _pretty big_ footgun

Reply:
Yes, it was happening here previously. Now that we have changed `StructColumn` to return a shallow copy always, this line would return a shallow copy of the children in line 6536, which then would result in mutations to the child columns propagating to the struct column here. see this example:

```python
In [1]: import cudf

In [2]: df = cudf.DataFrame({'a': [1, 2, 3], 'b':[10, 11, 12]})

In [3]: s = df.to_struct()

In [4]: s
Out[4]: 
0    {'a': 1, 'b': 10}
1    {'a': 2, 'b': 11}
2    {'a': 3, 'b': 12}
dtype: struct

In [5]: df.a[0]=10

In [6]: df
Out[6]: 
    a   b
0  10  10
1   2  11
2   3  12

In [7]: s
Out[7]: 
0    {'a': 10, 'b': 10}
1     {'a': 2, 'b': 11}
2     {'a': 3, 'b': 12}
dtype: struct
```

To avoid this introducing this behavior I'm explicitly making deep copies of child columns in line 6538(on the right) and then creating a struct column.